DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeIDGenerator.swift

```diff
@@ -33,12 +33,16 @@ internal final class NodeIDGenerator {
     /// Returns single `NodeID` for given instance of `UIView`.
     /// - Parameter view: the `UIView` object
     /// - Returns: the `NodeID` of queried instance
-    func nodeID(for view: UIView) -> NodeID {
-        if let currentID = view.nodeID {
+    func nodeID(view: UIView, nodeRecorder: NodeRecorder) -> NodeID {
```

Comment:
looks good, you may want to update the docs with new param.

same for below method.